app/components/TokenTable/TableRow.tsx

```diff
@@ -0,0 +1,179 @@
+import Link from "next/link";
+import React from "react";
+import PriceChangeContainer from "./PriceChangeContainer";
+import { Line } from "react-chartjs-2";
+import { Chart as ChartJS } from "chart.js/auto";
+import { CategoryScale, LinearScale, LineElement } from "chart.js";
+
+ChartJS.register(CategoryScale, LinearScale, LineElement);
+
+interface TableRowProps {
+  token: TokenInfo;
+}
+
+interface TokenInfo {
+  id: string;
+  market_cap_rank: number;
+  image: string;
+  symbol: string;
+  name: string;
+  current_price: number;
+  price_change_percentage_1h_in_currency: number;
+  price_change_percentage_24h_in_currency: number;
+  price_change_percentage_7d_in_currency: number;
+  total_volume: number;
+  market_cap: number;
+  circulating_supply: number;
+  total_supply: number;
+  sparkline_in_7d: Sparkline;
+}
+
+interface Sparkline {
+  price: number[];
+}
+
+const TableRow = ({ token }: TableRowProps) => {
+  const formatNum = (num: number): string => {
+    if (num >= 1_000_000_000_000) {
+      return `${(num / 1_000_000_000_000).toFixed(2)}T`;
+    } else if (num >= 1_000_000_000) {
+      return `${(num / 1_000_000_000).toFixed(2)}B`;
+    } else if (num >= 1_000_000) {
+      return `${(num / 1_000_000).toFixed(2)}M`;
+    } else {
+      return num === null ? "NaN" : num.toLocaleString();
+    }
+  };
```

Comment:
this function shou;d be in a utils folder, because you will end up using it again